src/App.tsx

```diff
@@ -15,36 +16,84 @@
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  function sortGoodsList(goods: string[], sortBy: string, isReverse: boolean) {
+    const sortedList = [...goods].sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+
+    return isReverse ? sortedList.reverse() : sortedList;
+  }
+
+  const handleReset = () => {
+    setReversed(false);
+    setSortType('');
+  };
+
+  const visibleGoods = sortGoodsList(goodsFromServer, sortType, reversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortType(SortType.alphabet)}
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortType !== SortType.alphabet,
+          })}
+        >
           Sort alphabetically
         </button>
-
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortType(SortType.length)}
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortType !== SortType.length,
+          })}
+        >
           Sort by length
         </button>
-
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setReversed(!reversed)}
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+        >
           Reverse
         </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortType.length || reversed) && (
+          <button
+            onClick={handleReset}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map((good, index) => (
```

Comment:
[CODE KNOWLEDGE] - When rendering a list, don't forget to add `key` prop. Using the index as a key is not recommended if the list can change because it may negatively impact performance and cause issues with component state. Instead, use a unique identifier like `good` if it's unique, or another unique property from the items.